design-toolkit/components/src/components/hero/index.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { Icon } from '@/index';
+import type { PropsWithChildren, ReactElement } from 'react';
+import { Children, createContext, isValidElement, useContext } from 'react';
+import type { ContextValue } from 'react-aria-components';
+import { type HeroStyleVariants, HeroStyles } from './styles';
+
+interface ChildProps extends PropsWithChildren {
+  className?: string;
+}
+
+interface HeroProps extends HeroStyleVariants, PropsWithChildren {
+  className?: string;
+}
+
+export const HeroContext =
+  createContext<ContextValue<HeroProps, HTMLDivElement>>(null);
+
+export function Hero({ children, className, ...props }: HeroProps) {
+  const slots = {
+    icon: null as ReactElement | null,
+    primary: null as ReactElement | null,
+    secondary: [] as ReactElement[],
+    invalid: [] as ReactElement[],
+  };
+  const styles = HeroStyles({ ...props });
+
+  Children.forEach(children, (child) => {
+    if (isValidElement(child)) {
+      const childType = child.type as { displayName?: string };
+      const displayName = childType?.displayName;
+
+      if (displayName === 'Hero.Icon') {
+        slots.icon = child;
+      } else if (displayName === 'Hero.Primary') {
+        slots.primary = child;
+      } else if (displayName === 'Hero.Secondary') {
+        slots.secondary.push(child);
+      } else {
+        slots.invalid.push(child);
+      }
+    }
+  });
+
+  return (
+    <HeroContext.Provider value={props}>
+      <header className={styles.container({ className })}>
+        <aside>{slots.icon}</aside>
+        <main>
+          {slots.primary}
+          {slots.secondary}
+
+          {slots.invalid.length > 0 && (
+            <div className='mt-l rounded-large bg-[hotPink] p-l'>
+              <h1 className='text-header-l'>Invalid {'<Hero />'} content</h1>
+              {slots.invalid}
+            </div>
+          )}
+        </main>
+      </header>
+    </HeroContext.Provider>
+  );
+}
+
+Hero.Icon = HeroIcon;
+Hero.Primary = HeroPrimary;
+Hero.Secondary = HeroSecondary;
+
+function HeroIcon({ children, className }: ChildProps) {
+  const { compact, theme } = useHeroVariants();
+  const styles = HeroStyles({ compact, theme });
+
+  return <Icon className={styles.icon({ className })}>{children}</Icon>;
```

Comment:
You can eliminate this component by implementing the `Icon.Provider` to pass down the className from the Hero

Reply:
Do you have an example of this? I am not sure what you are hinting at here.

Reply:
https://github.com/gohypergiant/standard-toolkit/blob/main/design-toolkit/components/src/components/avatar/index.tsx#L50

https://github.com/gohypergiant/standard-toolkit/blob/main/design-toolkit/components/src/components/avatar/index.tsx#L71

https://github.com/gohypergiant/standard-toolkit/pull/369/files#diff-7f04bd9f183f0749f1410b8f3bd50df42559a2287cd56db9a2176eded2aa4e9fR117

https://github.com/gohypergiant/standard-toolkit/pull/370#discussion_r2226865176

Reply:
I think I've got this now.